core/src/main/scala/com/microsoft/azure/synapse/ml/logging/fabric/FabricTokenParser.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.fabric
+
+import spray.json._
+
+class InvalidJwtTokenException(message: String) extends Exception(message)
+
+class JwtTokenExpiryMissingException(message: String) extends Exception(message)
+
+class FabricTokenParser(JWToken: String) {
+  private val tokens: Array[String] = JWToken.split("\\.")
+  private val parsedToken: JsValue = tokenCheckAndDecode(Some(tokens))
+
+  def getExpiry: Long = {
+    val exp: Option[Long] = parsedToken.asJsObject.fields.get("exp").collect { case JsNumber(value) => value.toLong }
+    exp match {
+      case Some(expValue) =>
+        expValue
+      case None =>
+        throw new JwtTokenExpiryMissingException(s"JWT token does not have expiration set. " +
+          s"Here is the token = {$JWToken}")
+    }
+  }
+
+  private def tokenCheckAndDecode(tokens: Option[Array[String]]): JsValue = {
```

Comment:
If this isnt still used please remove. Also once you remove this some other dependant tests can likely aslso go

Reply:
This and dependent tests removed.